pkg/lockfile/fixtures/pip/with-multiple-r-options.txt

```diff
@@ -0,0 +1,5 @@
+-r ./one-package-constrained.txt
+-r ./multiple-packages-mixed.txt
+
+# this should take precedents as it comes after
```

Comment:
I don't think this comment is correct, even if it goes before pip will still install 1.2.3.
Pip doesn't allow conflicting requirements, and if not given exact versions, will try to find the version that satisfies  all requirements if it can, order does not seem to matter.

Reply:
hmm ok so then in this case Pip would error? I guess the easiest path forward would be to just treat them as separate packages for now, and then this can be improved in future if needed (especially since this is the existing behaviour regardless of if `-r` is present)

Reply:
In this case pip will install `1.2.3` because `requests` in `./multiple-packages-mixed.txt` doesn't have a version constraint.

Yeah I think treating them as separate packages is probably fine for now.

Reply:
yeah sorry you're right, I was thinking of the original test case I wrote which was using `django` (which is contrained in `one-package-constrained.txt`)